keras_cv/layers/preprocessing/fmix.py

```diff
@@ -0,0 +1,207 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class FMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """FMix implements the FMix data augmentation technique.
+
+    Args:
+        alpha: Float value for beta distribution.  Inverse scale parameter for the gamma
+            distribution.  This controls the shape of the distribution from which the
+            smoothing values are sampled.  Defaults 0.5, which is a recommended value
+            in the paper.
+        decay_power: A float value representing the decay power for frequency decay prop
+            1/f**d.  Defaults to 3, as recommended in the paper.
+        max_soft: Float value representing softening value between 0 and 0.5 which
+            smooths hard edges in the mask.  Defaults to 0.0.
+        seed: Integer. Used to create a random seed.
+    References:
+        [FMix paper](https://arxiv.org/abs/2002.12047).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    fmix = keras_cv.layers.preprocessing.mix_up.FMix(0.5)
+    augmented_images, updated_labels = fmix({'images': images, 'labels': labels})
+    # output == {'images': updated_images, 'labels': updated_labels}
+    ```
+    """
+
+    def __init__(self, alpha=0.2, decay_power=3, max_soft=0.0, seed=None, **kwargs):
+        super().__init__(seed=seed, **kwargs)
+        self.alpha = alpha
+        self.decay_power = decay_power
+        self.max_soft = max_soft
+        self.seed = seed
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta, shape):
+        sample_alpha = tf.random.gamma(shape, 1.0, beta=alpha)
+        sample_beta = tf.random.gamma(shape, 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def fftfreq(self, n, d=1):
+        # This function returns the sample frequencies of a discrete fourier transform.
+
+        N = tf.convert_to_tensor((n - 1) / 2 + 1)
+        results = tf.concat(
+            [tf.range(N, dtype=tf.int32), tf.range(-(n // 2), 0, dtype=tf.int32)], 0
+        )
+        return results / (n * d)
+
+    def apply_fftfreq(self, h, w):
+        # Applying the fourier transform across 3 dimensions.
+
+        fx = self.fftfreq(w)[: w // 2 + 1 + w % 2]
+
+        fy = self.fftfreq(h)
+        fy = tf.expand_dims(fy, -1)
+
+        return tf.math.sqrt(fx * fx + fy * fy)
+
+    def get_spectrum(self, freqs, decay_power, ch, h, w):
+        # Function to apply a low pass filter by decaying its high frequency components.
+
+        scale = tf.ones(1) / tf.cast(
+            tf.math.maximum(freqs, tf.convert_to_tensor([1 / tf.reduce_max([w, h])]))
+            ** decay_power,
+            tf.float32,
+        )
+
+        param_size = tf.concat(
+            [tf.constant([ch]), tf.shape(freqs), tf.constant([2])], 0
```

Comment:
lets expand some of the abbreviations where possible.  `channel` is easier to read than `ch`